pkg/segment/reader/segread/segstatsreader.go

```diff
@@ -188,113 +188,139 @@ func readSingleSst(fdata []byte, qid uint64) (*structs.SegStats, error) {
 }
 
 func GetSegMin(runningSegStat *structs.SegStats,
-	currSegStat *structs.SegStats) (*utils.NumTypeEnclosure, error) {
+	currSegStat *structs.SegStats) (*utils.CValueEnclosure, error) {
 
-	rSst := utils.NumTypeEnclosure{
-		Ntype:    utils.SS_DT_SIGNED_NUM,
-		IntgrVal: math.MaxInt64,
-	}
+	result := utils.CValueEnclosure{}
 
 	if currSegStat == nil {
 		log.Errorf("GetSegMin: currSegStat is nil")
-		return &rSst, errors.New("GetSegMin: currSegStat is nil")
-	}
-
-	if !currSegStat.IsNumeric {
-		log.Errorf("GetSegMin: current segStats is non-numeric")
-		return &rSst, errors.New("GetSegMin: current segStat is non-numeric")
+		return &result, errors.New("GetSegMin: currSegStat is nil")
 	}
 
 	// if this is the first segment, then running will be nil, and we return the first seg's stats
 	if runningSegStat == nil {
+		if !currSegStat.IsNumeric && currSegStat.StringStats.Min.Dtype == utils.SS_DT_STRING {
+			return &utils.CValueEnclosure{
+				Dtype: utils.SS_DT_STRING,
+				CVal:  currSegStat.StringStats.Min.CVal,
+			}, nil
+		}
+
 		switch currSegStat.NumStats.Min.Ntype {
```

Comment:
nit: only do this switch statement if `IsNumeric`

Reply:
addressed in the latest commit.